edgelet/iotedge/src/init/import/old_config/mod.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+mod agent;
+pub(crate) use agent::*;
+
+mod moby_runtime;
+pub(crate) use moby_runtime::*;
+
+mod provisioning;
+pub(crate) use provisioning::*;
+
+use std::path::PathBuf;
+
+use url::Url;
+
+#[derive(Debug, serde_derive::Deserialize)]
+pub(crate) struct Config {
+    pub(crate) provisioning: Provisioning,
```

Comment:
if the struct is `pub(crate)`, marking individual fields as `pub(crate)` as opposed to just `pub` isn't necessary.

ditto for every other instance of this.

Reply:
I think it's better to be precise. The field is only written within the crate, so the field should be `pub(crate)`. Otherwise it's a slippery slope - if a `mod` is `pub(crate)` should all the items in the `mod` be made `pub` ?

Reply:
I mean... yeah, that's what I do 😶
But whatever, it's not a dealbreaker or anything.

Reply:
Feels like a footgun. You're one refactor away from accidentally increasing your API surface when you didn't mean to.

Reply:
I mean, when I write libraries, I tend to keep an eye on the `cargo doc` output, which makes those cases pretty obvious...

In any case, I wrote out some more thoughts on the matter, but after reading them over I feel like it kinda just boils down to personal preference, and idk if there's a clear best answer. My gut feeling is that `pub(crate)` should stay at the item level, not really the field level, but like I said, that's just a gut feeling.